cmd/nerdctl/login.go

```diff
@@ -287,7 +289,14 @@ func tryLoginWithRegHost(ctx context.Context, rh docker.RegistryHost) error {
 		}
 		res, err := ctxhttp.Do(ctx, rh.Client, req)
 		if err != nil {
-			return fmt.Errorf("failed to call rh.Client.Do: %w", err)
+			// In some circumstance (e.g. people just use 80 port to support pure http), the error will contain message like "dial tcp <port>: connection refused"
+			if (imgutil.IsErrHTTPResponseToHTTPSClient(err) || imgutil.IsErrConnectionRefused(err)) && insecure {
```

Comment:
If the registry service does not support HTTPS and the user does not specify `--insecure-registry`, we should alert them, like https://github.com/containerd/nerdctl/blob/v0.16.1/cmd/nerdctl/push.go#L197-L201